lib/prepare.js

```diff
@@ -814,6 +842,16 @@ function updateIconResourceForAdaptive (preparedIcons, resourceMap, platformReso
             defaultTargetPathForeground = getAdaptiveImageResourcePath(platformResourcesDir, 'mipmap', 'mdpi', 'ic_launcher_foreground.png', path.basename(default_icon.foreground));
             resourceMap[defaultTargetPathForeground] = default_icon.foreground;
         }
+
+        if (path.extname(path.basename(monochrome)) === '.xml') {
+            // Vector Use Case
+            defaultTargetPathMonochrome = getAdaptiveImageResourcePath(platformResourcesDir, 'mipmap', 'mdpi', 'ic_launcher_monochrome.xml', path.basename(default_icon.monochrome));
+            resourceMap[defaultTargetPathMonochrome] = default_icon.monochrome;
+        } else if (path.extname(path.basename(monochrome)) === '.png') {
+            // Images Use Case
+            defaultTargetPathMonochrome = getAdaptiveImageResourcePath(platformResourcesDir, 'mipmap', 'mdpi', 'ic_launcher_monochrome.png', path.basename(default_icon.monochrome));
+            resourceMap[defaultTargetPathMonochrome] = default_icon.monochrome;
+        }
```

Comment:
I think this also needs to be contained in a `if (monochrome)` block for the same reasons above.